libmamba/src/core/url.cpp

```diff
@@ -15,6 +17,152 @@
 
 namespace mamba
 {
+
+    namespace
+    {
+        /**
+         * A RAII `CURL*` createed from `curl_easy_handle`.
+         *
+         * Never null, throw exception at construction if creating the handle fails.
+         */
+        class CurlEasyHandle
+        {
```

Comment:
There is already such a handle in curl.hpp that could be used instead.

Reply:
There are some differences:
 - `~CurlHandle` calls ` curl_slist_free_all(p_headers)` which was not the case here.
 - There is no `CURL* raw();` which was the point here.
 
 Overall `curl.hpp` seems to be intended for actually using CURL to connect to the Internet, whereas here it is only a implementation detail for parsing urls (or at least what I'm trying to move it forward)

Reply:
If we keep that new RAII type, please add a comment over the class to explain why this exist and how it is different from the CurlHandle, like you did here.

Reply:
- freeing the p_headers is not  an issue, the pointer is null constructed so it is a noop if you don't explicitely add any header.
- curl should wrapped instead of beeing exposed through a raw accessor, thus the usage of the wrapper in curl.


Anyway, if you prefer to keep.this tony wrapper (which is not a problem asnlong as it is not exposed in any header) you need to implement proper valu semantics; the current implementation has the same issue as CurlStr regarding the destructor.